node/peer.js

```diff
@@ -207,4 +240,33 @@ TChannelPeer.prototype.makeOutConnection = function makeOutConnection(socket) {
     return conn;
 };
 
+TChannelPeer.prototype.pendingWeightedRandom = function pendingWeightedRandom() {
+    // A weighted random variable:
+    //   random() ** (1 / weight)
+    // Such that the probability distribution is uniform for weights of 0, but
+    // an increasing bias with increasing weight.
+    // However, although weight should start at 0 and increase probability,
+    // the number of pending requests starts at 0 and should decrease
+    // probability as it increases.
+    // For 0 pending requests, we produce a uniform probability distribution by
+    // raising a uniform random variable to the power of 1 (pending + 1).
+    // As the number of pending requests increase, the magnitude of the power
+    // increases and the probability distribution develops a bias toward zero.
+    // TODO review weighted reservoir sampling:
+    // http://arxiv.org/pdf/1012.0256.pdf
+    var self = this;
+    return Math.pow(self.random(), 1 + self.pending);
+};
+
+TChannelPeer.prototype.createStat = function createStat(name, type, value) {
```

Comment:
This is not needed; it does not anything except add `self.peer` which is already accessible since you do `peer.statThing.onwat()`.

Get rid of this indirection.